kafka-workitem/src/main/java/org/jbpm/process/workitem/kafka/KafkaWorkItemHandler.java

```diff
@@ -0,0 +1,123 @@
+                   @WidParameter(name = "ValueSerializerClass", required = true),
+                   @WidParameter(name = "Topic", required = true),
+                   @WidParameter(name = "Key", required = true),
+                   @WidParameter(name = "Value", required = true)
+     },
+     results = {
+                @WidResult(name = "Result")
+     },
+     mavenDepends = {
+                     @WidMavenDepends(group = "${groupId}", artifact = "${artifactId}", version = "${version}")
+     },
+     serviceInfo = @WidService(category = "${name}", description = "${description}",
+                               keywords = "kafka",
+                               action = @WidAction(title = "Publish message to a kafka topic")
+     ))
+
+public class KafkaWorkItemHandler extends AbstractLogOrThrowWorkItemHandler {
+
+    private static final Logger LOG = LoggerFactory.getLogger(KafkaWorkItemHandler.class);
+
+    private Producer<Long, String> producer;
+
+    public void executeWorkItem(WorkItem workItem, WorkItemManager manager) {
+
+        Map<String, Object> results = new HashMap<String, Object>();
+
+        String bootstrapServers = (String) workItem.getParameter("BootstrapServers");
+        String client_id = (String) workItem.getParameter("ClientId");
+        String keySerializerClass = (String) workItem.getParameter("KeySerializerClass");
+        String valueSerializerClass = (String) workItem.getParameter("ValueSerializerClass");
+        String topic = (String) workItem.getParameter("Topic");
+        String key = (String) workItem.getParameter("Key");
+        String value = (String) workItem.getParameter("Value");
+
+
+        Properties config = new Properties();
+        config.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers);
+        config.put(ProducerConfig.CLIENT_ID_CONFIG, client_id);
+        config.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, keySerializerClass);
+        config.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, valueSerializerClass);
+
+        if (producer == null) {
+            producer = createProducer(config);
+        }
+        try {
+
+            producer.send(new ProducerRecord(topic, key, value));
+            results.put("Result", "success");
+            manager.completeWorkItem(workItem.getId(), results);
+        } catch (Exception e) {
+            LOG.error("Kafka error", e);
+            producer.flush();
+            producer.close();
+            results.put("Result", "failure");
+            handleException(e);
+        }
+
+    }
+
+    public void abortWorkItem(WorkItem workItem, WorkItemManager manager) {
+
+    }
+
+    private static Producer<Long, String> createProducer(Properties config) {
+        return new KafkaProducer<Long, String>(config);
+    }
+
+    public Producer<Long, String> getProducer() {
+        return producer;
+    }
+
+    public void setProducer(Producer<Long, String> producer) {
```

Comment:
Don't use this method for tests only, pass in the mock producer as wid parameter, for example above:

```java
 Producer testProducer = (Producer) workItem.getParameter("TestProducer");
...and then below check for example...
if(testProducer != null) {
producer = testProducer
} else {
producer = createProducer(config);
}

```

Reply:
Implemented Cacheable and moved producer creation inside constructor. But I have left the setProducer(..) unchanged. IMHO, it is not a good idea to include a test producer(MockProducer) in WIH main class either through a wid parameter or through a constructor. Please suggest if this change is really needed or not.